pkg/daemon/controller.go

```diff
@@ -181,12 +177,69 @@ func (c *Controller) processNextSubnetWorkItem() bool {
 	return true
 }
 
-func (c *Controller) reconcileRouters() error {
+func (c *Controller) reconcileRouters(event subnetEvent) error {
 	subnets, err := c.subnetsLister.List(labels.Everything())
 	if err != nil {
 		klog.Errorf("failed to list namespace %v", err)
 		return err
 	}
+
+	var ok bool
+	var oldSubnet, newSubnet *kubeovnv1.Subnet
+	if event.old != nil {
+		if oldSubnet, ok = event.old.(*kubeovnv1.Subnet); !ok {
+			klog.Errorf("expected old subnet in subnetEvent but got %#v", event.old)
+			return nil
+		}
+	}
+	if event.new != nil {
+		if newSubnet, ok = event.new.(*kubeovnv1.Subnet); !ok {
+			klog.Errorf("expected new subnet in subnetEvent but got %#v", event.new)
+			return nil
+		}
+	}
+
+	// handle policy routing
+	rulesToAdd, rulesToDel, routesToAdd, routesToDel, err := c.diffPolicyRouting(oldSubnet, newSubnet)
+	if err != nil {
+		klog.Errorf("failed to get policy routing difference: %v", err)
+		return err
+	}
+	// add new routes first
+	for _, r := range routesToAdd {
+		if err = netlink.RouteAdd(&r); err != nil && !errors.Is(err, syscall.EEXIST) {
```

Comment:
what will happen, if the route already exists in host

Reply:
`!errors.Is(err, syscall.EEXIST)` ensures the existing route will not be replaced.

Reply:
> what will happen, if the route already exists in host

@oilbeater  `!errors.Is(err, syscall.EEXIST)` ensures the existing route will not be replaced.